x-pack/plugins/upgrade_assistant/server/routes/status.test.ts

```diff
@@ -76,21 +144,43 @@ describe('Status API', () => {
       expect(resp.payload).toEqual({
         readyForUpgrade: false,
         details:
-          'You have 1 Elasticsearch deprecation issue and 1 Kibana deprecation issue that must be resolved before upgrading.',
+          'You have 1 system index that must be migrated and ' +
+          '1 Elasticsearch deprecation issue and 1 Kibana deprecation issue that must be resolved before upgrading.',
       });
     });
 
-    it('returns readyForUpgrade === true if there are no critical deprecations', async () => {
-      ESUpgradeStatusApis.getESUpgradeStatus.mockResolvedValue({
-        cluster: [],
-        indices: [],
+    it('returns readyForUpgrade === false if no critical Kibana or ES deprecations but system indices need migration', async () => {
+      getESUpgradeStatusMock.mockResolvedValue(esNoDeprecationsResponse);
+
+      getKibanaUpgradeStatusMock.mockResolvedValue({
         totalCriticalDeprecations: 0,
       });
 
-      KibanaUpgradeStatusApis.getKibanaUpgradeStatus.mockResolvedValue({
+      getESSystemIndicesMigrationStatusMock.mockResolvedValue(systemIndicesMigrationResponse);
+
+      const resp = await routeDependencies.router.getHandler({
+        method: 'get',
+        pathPattern: '/api/upgrade_assistant/status',
+      })(routeHandlerContextMock, createRequestMock(), kibanaResponseFactory);
+
+      expect(resp.status).toEqual(200);
+      expect(resp.payload).toEqual({
+        readyForUpgrade: false,
+        details:
+          'You have 1 system index that must be migrated and ' +
+          '0 Elasticsearch deprecation issues and 0 Kibana deprecation issues that must be resolved before upgrading.',
```

Comment:
nit: probably better to address this separately since the logic was here from before and not so relevant to this pr, but it seems to me would be better to just bring to the user what actually needs to be addressed; the `0 es and 0 kibana deprecations that need to be resolved` might not be relevant information for our users